File: src/main/kotlin/basic_control_structures/exercise_9.kt
package basic_control_structures

// Write an algorithm that reads how much money a person has in their wallet (in R$) and shows how many dollars they can buy. Consider US$1.00 = R$3.45.
fun moneyConversionRealToDollar() {
    print("Amount: R$")
    val amount: Double = readln().toDouble()
    val _amount = String.format("%.2f", amount)

    print("Exchange rate: R$")
    val price: Double = readln().toDouble()
    val _price = String.format("%.2f", price)

    val dollar = amount / price
    val _dollar = String.format("%.2f", dollar)

    println("You have R$$_amount, the exchange rate is R$$_price and you can buy US$$_dollar.")
}

File: Dart/a_basic_control_structure/exercise_9.dart
// Write an algorithm that reads how much money a person has in their wallet (in R$) and shows how many dollars they can buy. Consider US$1.00 = R$3.45.

import 'dart:io';

void main() {
  double exchangeRate = 3.45;
  double amount = getAmount();

  buyDollar(amount, exchangeRate);
}

double getAmount() {
  while (true) {
    stdout.write('Amount: R\$');
    String? input = stdin.readLineSync();

    if (input != null) {
      try {
        return double.parse(input);
      } catch (exception) {
        print('Error: Please enter a valid integer');
      }
    }
  }
}

void buyDollar(double amount, double exchangeRate) {
  String _amount = amount.toStringAsFixed(2);

  double dollar = amount / exchangeRate;
  String _dollar = dollar.round().toStringAsFixed(2);

  print(
      'You have R\$$_amount, the exchange rate is R$exchangeRate and you can buy US\$${_dollar}.');
}
